components/nexusoperations/frontend/handler.go

```diff
@@ -315,6 +339,10 @@ func (c *requestContext) interceptRequest(ctx context.Context, request *nexus.Co
 		return commonnexus.ConvertGRPCError(err, true)
 	}
 
-	// TODO: Apply other relevant interceptors.
+	if err := c.clientVersionChecker.ClientSupported(ctx); err != nil {
```

Comment:
Seems like there should be a separate method for supported nexus clients just to avoid mixing up the Temporal / Nexus verification logic but it's not critical.